### DISCUSSION TITLE
Correct Dokku storage settings to deploy Mautic

### DISCUSSION THREAD

**Discussion by FickleLife** [dokku/dokku#4588] [Q&A] [answered]
I'm attempting to set up Mautic (https://github.com/mautic/docker-mautic) on Dokku. I have everything working well except for the mounted volume. Mautic stores config files in the volume and if the volume is not present every time the container restarts it needs to be reconfigured (DB connection string, etc). The vanilla docker instructions on the above page are:

```
$ docker volume create mautic_data

$ docker run --name mautic -d \
    --restart=always \
    -e MAUTIC_DB_HOST=127.0.0.1 \
    -e MAUTIC_DB_USER=root \
    -e MAUTIC_DB_PASSWORD=mypassword \
    -e MAUTIC_DB_NAME=mautic \
    -e MAUTIC_RUN_CRON_JOBS=true \
    -e MAUTIC_TRUSTED_PROXIES=0.0.0.0/0 \
    -p 8080:80 \
    -v mautic_data:/var/www/html \
    mautic/mautic:latest
```

I have created a persistent volume in dokku with `dokku storage:mount mautic /var/lib/dokku/data/storage/mautic:/mautic_data`, this is confirmed:

```
root@apps:/var/lib# dokku storage:report mautic
=====> mautic storage information
       Storage build mounts:                                   
       Storage deploy mounts:         -v /var/lib/dokku/data/storage/mautic:/mautic_data 
       Storage run mounts:            -v /var/lib/dokku/data/storage/mautic:/mautic_data 
```

I have also created extra folders so that there is a path /var/lib/dokku/data/storage/mautic/var/www/html

However the config file is not saved, and therefore Mautic wants to be configured every time the container is restarted. Can anyone point out where I am going wrong?

**Comment by Cellane** [comment, marked as answer]
I think you’re understanding the volume mount statement from Mautic’s documentation wrong – it doesn’t mount the container’s `/mautic_data` folder into host’s `/var/www/html` directory, it’s the opposite – it mounts the container’s `/var/www/html` directory into a named mount `mautic_data` on host.

Thus what you should do, I think, is to run these three commands:

```bash
dokku storage:unmount mautic /var/lib/dokku/data/storage/mautic:/mautic_data
rm -rf /var/lib/dokku/data/storage/mautic # Optional but recommended
dokku storage:mount mautic /var/lib/dokku/data/storage/mautic:/var/www/html
```

In other words, to me it seems like Mautic is storing the configuration files under `/var/www/html` directory in the container.

**Reply by FickleLife** [reply]
Thank you @Cellane , that was the one. Appreciate it.
